src/backend/base/langflow/components/nvidia/nemo_evaluator.py

```diff
@@ -245,45 +317,52 @@ def add_inputs_with_saved_values(self, build_config, input_definitions, saved_va
             build_config.setdefault("_dynamic_fields", []).append(input_name)
 
     def add_evaluation_inputs(self, build_config, saved_values, evaluation_type):
-        """Adds inputs based on the evaluation type (LM Evaluation or Custom Evaluation)."""
+        """
+        Adds inputs based on the evaluation type (LM Evaluation or Custom Evaluation).
+        """
         if evaluation_type == "LM Evaluation Harness":
             self.add_inputs_with_saved_values(build_config, self.lm_evaluation_inputs, saved_values)
         elif evaluation_type == "Custom Evaluation":
             self.add_inputs_with_saved_values(build_config, self.custom_evaluation_inputs, saved_values)
+        elif evaluation_type == "LLM-as-a-Judge":
+            self.add_inputs_with_saved_values(build_config, self.llm_as_judge, saved_values)
 
     def update_build_config(self, build_config, field_value, field_name=None):
-        """Updates the component's configuration based on the selected option."""
+        """
+        Updates the component's configuration based on the selected option.
+        """
         try:
-            message = f"Updating build config: field_name={field_name}, field_value={field_value}"
-            logger.info(message)
+            print(f"Updating build config: field_name={field_name}, field_value={field_value}")
 
             saved_values = {}
 
             if field_name == "000_llm_name":
                 # Refresh model options for LLM Name dropdown
                 build_config["000_llm_name"]["options"] = self.fetch_models()
+                print("Updated LLM Name options:", build_config["000_llm_name"]["options"])
+
+            elif field_name == "judge_llm_name":
+                # Refresh model options for LLM Name dropdown
+                build_config["judge_llm_name"]["options"] = self.fetch_models()
+                print("Updated LLM Name options:", build_config["judge_llm_name"]["options"])
 
             elif field_name == "002_evaluation_type":
                 self.clear_dynamic_inputs(build_config, saved_values)
                 self.add_evaluation_inputs(build_config, saved_values, field_value)
-
             elif field_name == "310_run_inference":
```

Comment:
I don't think the changes in this elif clause are a good idea. 310_run_inference should only be used when custom evaluation is selected. My intent was that each evaluation type have a completely distinct set of sub-options, makes everything cleaner. This logic is getting too complex and will be hard to maintain.

Reply:
Agreed, that's how it is.